### ISSUE TITLE
Problem z PUT błąd "0"

### ISSUE THREAD

**Issue by lukaszkrakowiak** [allegro/allegro-api#4466]
Dzień dobry, 
Nie mogę wywołać "PUT" na allegro.
Kod jaki korzystam:
`$link = "https://api.allegro.pl/sale/offers/".$offerId;
        $ch = curl_init($link);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
        curl_setopt($ch, CURLOPT_PUT, 1);
        curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);
        curl_setopt($ch, CURLOPT_HTTPHEADER, [
            "Authorization: Bearer " . $this->token,
            "Accept: application/vnd.allegro.public.v1+json",
            "Content-Type: application/vnd.allegro.public.v1+json"
        ]);`

jako $payload jest wartość pobrana i zmodyfikowana z zasobu "https://api.allegro.pl/sale/offers/{id}"

Zwracany kod błędu to "0"

**Comment by TimothyKoval** [comment]
Stawiam czteropak na problem z certyfikatami root po stronie klienta

**Comment by kfijalko** [comment]
Z tego kodu nic nie wynika.
Pokaż pełny request / response (skasuj token).

**Comment by lukaszkrakowiak** [comment]
@TimothyKoval może to być spowodowane działaniem na localhoscie?

**Comment by TimothyKoval** [comment]
@lukaszkrakowiak Niekoniecznie, ale zakładam, że korzystasz ze środowiska typu WAMP. Wygugluj jak zaktualizować w tym środowisku certyfikaty CA

**Comment by lukaszkrakowiak** [comment]
@TimothyKoval tak, dokładniej XAMP. Dobra to wrzuce na serwer i tam sprawdze :)

**Comment by lukaszkrakowiak** [comment]
@TimothyKoval na serwerze jest problem z "ServiceTimeoutException"

**Comment by TimothyKoval** [comment]
Daj `curl_getinfo`, `curl_error` i `curl_errno`

**Comment by lukaszkrakowiak** [comment]
//zaputanie

* Hostname api.allegro.pl was found in DNS cache
*   Trying 5.134.213.11...
* Connected to api.allegro.pl (5.134.213.11) port 443 (#0)
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: none
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* Server certificate:
*  subject: C=PL; ST=Wielkopolskie; L=Poznan; OU=IT; O=Allegro.pl sp. z o.o.; CN=edge.allegro.pl
*  start date: Apr 30 08:03:25 2020 GMT
*  expire date: Jun 28 13:20:05 2021 GMT
*  subjectAltName: host "api.allegro.pl" matched cert's "api.allegro.pl"
*  issuer: C=BE; O=GlobalSign nv-sa; CN=GlobalSign RSA OV SSL CA 2018
*  SSL certificate verify ok.
> PUT /sale/offers/id_oferty HTTP/1.1
Host: api.allegro.pl
Authorization: Bearer {token}
Accept: application/vnd.allegro.public.v1+json
Content-Type: application/vnd.allegro.public.v1+json
Content-Length: 5666
Expect: 100-continue

< HTTP/1.1 100 Continue
< HTTP/1.1 504 Gateway Timeout
< trace-id: 3723fa577e576652
< content-type: application/json; charset=utf-8
< content-length: 167
< date: Thu, 06 May 2021 10:01:15 GMT
* HTTP error before end of send, stop sending
< 
* Closing connection 0



//curl_getinfo
    [url] => https://api.allegro.pl/sale/offers/10514369472
    [content_type] => application/json; charset=utf-8
    [http_code] => 504
    [header_size] => 190
    [request_size] => 1400
    [filetime] => -1
    [ssl_verify_result] => 0
    [redirect_count] => 0
    [total_time] => 30.14382
    [namelookup_time] => 2.6E-5
    [connect_time] => 0.026705
    [pretransfer_time] => 0.086541
    [size_upload] => 0
    [size_download] => 167
    [speed_download] => 5
    [speed_upload] => 0
    [download_content_length] => 167
    [upload_content_length] => 5666
    [starttransfer_time] => 30.143806
    [redirect_time] => 0
    [redirect_url] => 
    [primary_ip] => 185.31.27.150
    [certinfo] => Array
        (
        )

    [primary_port] => 443
    [local_ip] => 138.201.62.218
    [local_port] => 54543

**Comment by lukaszkrakowiak** [comment]
@AureliuszBrussy czy moge prosić o pomoc?

**Comment by AureliuszBrussy** [comment]
@lukaszkrakowiak konsultuję Twoje zgłoszenie z odpowiednim zespołem. Wrócę z odpowiedzią.

**Comment by AureliuszBrussy** [comment]
@lukaszkrakowiak czy możesz przesłać nam cały request, bez wycinania id oferty i body (ale bez access tokena)? 

Jeśli nie chcesz tego zrobić tutaj na forum, możesz przesłać te dane przez [formularz kontaktowy](https://allegro.pl/pomoc/kontakt?kategoria=a0f7ce34-c6f6-43e6-af7d-ad39bacce619&subjectId=517305f2-5ea1-4b3b-9080-d4cc960ab6a1), wskaż tylko w zgłoszeniu numer wątku na GitHubie: #4466.

**Comment by lukaszkrakowiak** [comment]
Wysłałem poprzez formularz kontaktowy
